### PR TITLE
Upgrade to CircleCI 2.1

### PR DESCRIPTION
**Pull request by aldesantis**
**Description**

Upgrades to CircleCI 2.1, which allows us to use Orbs. This is especially important for https://github.com/solidusio/solidus/pull/3252, since we want to use [this Orb](https://github.com/nebulab/circleci-orbs-stoplight) for keeping the OAS documentation in sync with Stoplight.

**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [x] I have updated Guides and README accordingly to this change (if needed)
- [x] I have added tests to cover this change (if needed)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
       - run: |
           bundle install --path=vendor/bundle
 
-      - cache-save:
+      - save_cache:
```
Comment (kennyadsl) [inline .circleci/config.yml]: I'm sure there's a valid reason but can you explain this change better in the commit message, please?

Comment (aldesantis) [reply .circleci/config.yml]: @kennyadsl `cache-save` is not a valid step name in CircleCI, I'm actually not sure how this has worked so far. The right name is `save_cache`.

Comment (kennyadsl) [reply .circleci/config.yml]: 😱

I'm also noticing that we are trying to restore the key `solidus-gems-v1`, which is never saved...

Comment (aldesantis) [reply .circleci/config.yml]: @kennyadsl that's actually fine, because it will look for any keys that start with `solidus-gems-v1`.

Comment (kennyadsl) [reply .circleci/config.yml]: Found some reference about the [`cache-save`](https://thoughtbot.com/blog/circleci-2-rails) step, maybe it's the CircleCI legacy way of calling that type of jobs.

**Pull request merged**